tests/testDegenerateTemplate.py

```diff
@@ -0,0 +1,145 @@
+pexLogging.Log(root, 'lsst.meas.deblender.symmetricFootprint',
+               pexLogging.Log.INFO)
+pexLogging.Log(root, 'lsst.meas.deblender.getSignificantEdgePixels',
+               pexLogging.Log.INFO)
+pexLogging.Log(root, 'afw.Mask',
+               pexLogging.Log.INFO)
+
+
+def imExt(img):
+    bbox = img.getBBox()
+    return [bbox.getMinX(), bbox.getMaxX(),
+            bbox.getMinY(), bbox.getMaxY()]
+
+
+def doubleGaussianPsf(W, H, fwhm1, fwhm2, a2):
+    return measAlg.DoubleGaussianPsf(W, H, fwhm1, fwhm2, a2)
+
+
+def gaussianPsf(W, H, fwhm):
+    return measAlg.DoubleGaussianPsf(W, H, fwhm)
+
+
+class DegenerateTemplateTestCase(lsst.utils.tests.TestCase):
+
+    def test1(self):
+        '''
+        A simple example: three overlapping blobs (detected as 1
+        footprint with three peaks).  Additional peaks are added near
+        the blob peaks that should be identified as degenerate.
+        '''
+        H, W = 100, 100
+
+        fpbb = afwGeom.Box2I(afwGeom.Point2I(0, 0),
+                             afwGeom.Point2I(W-1, H-1))
+
+        afwimg = afwImage.MaskedImageF(fpbb)
+        imgbb = afwimg.getBBox()
+        img = afwimg.getImage().getArray()
+
+        var = afwimg.getVariance().getArray()
+        var[:, :] = 1.
+
+        blob_fwhm = 10.
+        blob_psf = doubleGaussianPsf(99, 99, blob_fwhm, 2.*blob_fwhm, 0.03)
+
+        fakepsf_fwhm = 3.
+        fakepsf = gaussianPsf(11, 11, fakepsf_fwhm)
+
+        blobimgs = []
+        x = 75.
+        XY = [(x, 35.), (x, 65.), (50., 50.)]
+        flux = 1e6
+        for x, y in XY:
+            bim = blob_psf.computeImage(afwGeom.Point2D(x, y))
+            bbb = bim.getBBox()
+            bbb.clip(imgbb)
+
+            bim = bim.Factory(bim, bbb)
+            bim2 = bim.getArray()
+
+            blobimg = np.zeros_like(img)
+            blobimg[bbb.getMinY():bbb.getMaxY()+1,
+                    bbb.getMinX():bbb.getMaxX()+1] += flux * bim2
+            blobimgs.append(blobimg)
+
+            img[bbb.getMinY():bbb.getMaxY()+1,
+                bbb.getMinX():bbb.getMaxX()+1] += flux * bim2
+
+        # Run the detection code to get a ~ realistic footprint
+        thresh = afwDet.createThreshold(5., 'value', True)
+        fpSet = afwDet.FootprintSet(afwimg, thresh, 'DETECTED', 1)
+        fps = fpSet.getFootprints()
+        print('found', len(fps), 'footprints')
+
+        # Add new peaks near to the first peaks that will be degenerate
+        fp0 = fps[0]
+        for x, y in XY:
+            fp0.addPeak(x-10, y+6, 10)
+
+        for fp in fps:
+            print('peaks:', len(fp.getPeaks()))
+            for pk in fp.getPeaks():
+                print('  ', pk.getIx(), pk.getIy())
+
+
+        deb = deblend(fp0, afwimg, fakepsf, fakepsf_fwhm, verbose=True, removeDegenerateTemplates=True)
+
+        self.assertTrue(deb.peaks[3].degenerate)
+        self.assertTrue(deb.peaks[4].degenerate)
+        self.assertTrue(deb.peaks[5].degenerate)
+
```

Comment:
Because I implicitly put the last three peaks into the list by hand.  The deblending code does not change the order of peaks and all have lower flux than the actual peaks from the images.